bindings/go/src/ucx/request.go

```diff
@@ -64,9 +64,10 @@ func NewRequest(request C.ucs_status_ptr_t, callbackId uint64, immidiateInfo int
 			case UcpAmDataRecvCallback:
 				callback(ucpRequest, requestStatus, uint64(immidiateInfo.(C.size_t)))
 			}
-			if requestStatus != C.UCS_OK {
-				return ucpRequest, NewUcxError(requestStatus)
-			}
+		}
+		ucpRequest.Status = requestStatus
+		if requestStatus != UCS_OK {
```

Comment:
can we remove requestStatus variable altogether, and always use ucpRequest.Status instead?